### PR TITLE
Implement toast messages

### PR DESCRIPTION
**Pull request by milkdropper**
For lack of a better word, this PR implements toast notifications. For now they are displayed centered in the screen with a hard coded timeout of 2 seconds (I can't imagine a reason for this being configurable).

For immediate use: lock/unlock in messages are displayed in libprojectM and audio toggle in SDL will produce a useful message.

Obligatory demo video as POC here: https://streamable.com/4mgxx8. 

With fast transitions the feedback of locking is really helpful. Also audio toggle from SDL is implemented showing the device name on the screen as per feedback here https://github.com/projectM-visualizer/projectm/issues/350.

image

### CODE REVIEW

File: src/libprojectM/Renderer/Renderer.cpp [hunk 1]
```diff
@@ -76,10 +76,13 @@ Renderer::Renderer(int width, int height, int gx, int gy, BeatDetect* _beatDetec
 	title_fontURL(_titlefontURL), menu_fontURL(_menufontURL), presetURL(_presetURL)
 {
 	this->totalframes = 1;
-	this->lastTime = duration_cast<milliseconds>(system_clock::now().time_since_epoch());
-	this->currentTime = duration_cast<milliseconds>(system_clock::now().time_since_epoch());
+	this->lastTimeFPS = duration_cast<milliseconds>(system_clock::now().time_since_epoch());
+	this->currentTimeFPS = duration_cast<milliseconds>(system_clock::now().time_since_epoch());
+	this->lastTimeToast = duration_cast<milliseconds>(system_clock::now().time_since_epoch());
+	this->currentTimeToast = duration_cast<milliseconds>(system_clock::now().time_since_epoch());
```
Comment (revmischa) [inline src/libprojectM/Renderer/Renderer.cpp:82]: Maybe refactor `duration_cast<milliseconds>(system_clock::now().time_since_epoch())` to a function

File: src/libprojectM/Renderer/Renderer.hpp [hunk 2]
```diff
@@ -102,13 +108,28 @@ class Renderer
   std::string fps() const {
 		return m_fps;
   }
+
+  void setToastMessage(const std::string& theValue)
+  {
+    // Initialize counters
+	lastTimeToast= duration_cast<milliseconds>(system_clock::now().time_since_epoch());
```
Comment (revmischa) [inline src/libprojectM/Renderer/Renderer.hpp:115]: wild indentation here

**Comment by milkdropper** [comment]
- In the last commit I tried to address the indentation. I am using Visual Studio on Windows. Any idea how I can make it adapt to the existing indentation? In my view there is really inconsistent indentation and I would prefer to make everything the same. I assume Visual Studio could automate fixing this.

- I refactored all the timing for FPS and toast messages into functions.

- I tried to address a problem with displaying text (more below).

Currently if your toast, preset name, or status is larger than the draw window it will just overflow. See below:
image

The current commit will cut off text and make it obvious it was and overflow by adding "...". Except when it's multi-line then it will cut it off with some buffer. See below:
image

I also left behind (but commented out) option #2: instead of cutting off the text it scales it down. See below:
image

**Comment by milkdropper** [comment]
Not sure why the travis build is showing incomplete. It looks done/passed when you click details.

**Pull request merged**